### PR TITLE
docs: updating links to the directory and wording

### PR DESCRIPTION
**Pull request by ali-hamza-noor**
Signed-off-by: alnoor <alihamzanoor99@gmail.com>

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

File: docs/developer-guide/test-e2e.md [hunk 1]
```diff
@@ -1,21 +1,21 @@
 # E2E Tests
 
-The directory contains E2E tests and test applications. The test assume that Argo CD services are installed into `argocd-e2e` namespace or cluster in current context. One throw-away
-namespace `argocd-e2e***` is created prior to tests execute. The throw-away namespace is used as a target namespace for test applications.
+This [directory](../../test) contains E2E tests and test applications. The tests assume that Argo CD services are installed into `argocd-e2e` namespace or cluster in current context. A throw-away
```
Comment (rumstead) [inline docs/developer-guide/test-e2e.md:3]:
```suggestion
The test [directory](../../test) contains E2E tests and test applications. The tests assume that Argo CD services are installed into `argocd-e2e` namespace or cluster in current context. A throw-away
```

Comment (rumstead) [reply docs/developer-guide/test-e2e.md:3]: ^ Super nit.

File: docs/developer-guide/test-e2e.md [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-The `test/e2e/testdata` directory contains various Argo CD applications. Before test execution directory is copies into `/tmp/argocd-e2e***` temp directory and used in tests as a
+The [/test/e2e/testdata](../../test/e2e/testdata) directory contains various Argo CD applications. Before test execution directory is copies into `/tmp/argocd-e2e***` temp directory and used in tests as a
 Git repository via file url: `file:///tmp/argocd-e2e***`.
 
 ## Running Tests Locally
 
-1. Start the e2e version `make start-e2e` 
-1. Run the tests: `make test-e2e`
+1. Start the e2e version `make start-e2e`
+2. Run the tests: `make test-e2e`
```
Comment (rumstead) [inline docs/developer-guide/test-e2e.md:12]: Markdown will automatically increment when you do repeated 1s.
https://argo-cd.readthedocs.io/en/stable/developer-guide/test-e2e/#running-tests-locally

**Review by rumstead — APPROVED** [review]
LGTM

File: docs/developer-guide/test-e2e.md [hunk 3]
```diff
@@ -1,21 +1,21 @@
 # E2E Tests
 
-The directory contains E2E tests and test applications. The test assume that Argo CD services are installed into `argocd-e2e` namespace or cluster in current context. One throw-away
-namespace `argocd-e2e***` is created prior to tests execute. The throw-away namespace is used as a target namespace for test applications.
+The test [directory](../../test) contains E2E tests and test applications. The tests assume that Argo CD services are installed into `argocd-e2e` namespace or cluster in current context. A throw-away
+namespace `argocd-e2e***` is created prior to the execution of the tests. The throw-away namespace is used as a target namespace for test applications.
 
-The `test/e2e/testdata` directory contains various Argo CD applications. Before test execution directory is copies into `/tmp/argocd-e2e***` temp directory and used in tests as a
+The [/test/e2e/testdata](../../test/e2e/testdata) directory contains various Argo CD applications. Before test execution directory is copies into `/tmp/argocd-e2e***` temp directory and used in tests as a
```
Comment (crenshaw-dev) [inline docs/developer-guide/test-e2e.md:6]: The links to github directories will fail in readthedocs: https://argo-cd--20335.org.readthedocs.build/en/20335/developer-guide/test-e2e/

If you want to link to them, use the full GitHub URL.

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: docs/developer-guide/test-e2e.md (hunk 3 above)
Comment (ALIHAMZA99) [reply docs/developer-guide/test-e2e.md:6]: good point, let me update that

**Review by crenshaw-dev — APPROVED** [review]
Thanks!

**Pull request merged**